packages/relay-kit/src/packs/safe-4337/estimators/PimlicoFeeEstimator.ts

```diff
@@ -19,6 +26,26 @@ export class PimlicoFeeEstimator implements IFeeEstimator {
     }
   }
 
+  async getPaymasterEstimation({
+    userOperation,
+    paymasterUrl,
+    entryPoint,
+    sponsorshipPolicyId
+  }: EstimateSponsoredFeeFunctionProps): Promise<EstimateSponsoredGasData> {
+    const paymasterClient = new ethers.JsonRpcProvider(paymasterUrl, undefined, {
+      batchMaxCount: 1
+    })
+
+    const params = [userOperationToHexValues(userOperation), entryPoint]
+
+    const gasEstimate = await paymasterClient.send(
+      'pm_sponsorUserOperation',
+      sponsorshipPolicyId ? [...params, { sponsorshipPolicyId }] : params
```

Comment:
You add this rule inside the send function when the params array is created 2 lines above. Wouldn't make sense to do this check when creating the element?

`params` is not used later for anything else

Reply:
we can do something like this instead:

```typescript

    const params = sponsorshipPolicyId
      ? [userOperationToHexValues(userOperation), entryPoint, { sponsorshipPolicyId }]
      : [userOperationToHexValues(userOperation), entryPoint]

    const gasEstimate = await paymasterClient.send('pm_sponsorUserOperation', params)
    
```

Reply:
done!

Reply:
Duplicated two params, but for me it improved the legibility